test/hotspot/jtreg/runtime/Safepoint/TestAbortVMOnSafepointTimeout.java

```diff
@@ -23,78 +23,58 @@
-            return;
-        }
-
-        testWith(500, 500);
-    }
-
-    static int test_loop(int x) {
-        int sum = 0;
-        if (x != 0) {
-            // Long running loop without safepoint.
-            for (int y = 1; y < Integer.MAX_VALUE; ++y) {
-                if (y % x == 0) ++sum;
+            Integer waitTime = Integer.parseInt(args[0]);
+            WhiteBox wb = WhiteBox.getWhiteBox();
+            while (!wb.waitUnsafe(waitTime)) {
+                System.out.println("Waiting for safepoint");
             }
+            System.out.println("This message would occur after some time.");
+        } else {
+            testWith(50, 1, 999);
         }
-        return sum;
     }
 
-    public static void testWith(int sfpt_interval, int timeout_delay) throws Exception {
-        // -XX:-UseCountedLoopSafepoints - is used to prevent the loop
-        // in test_loop() to poll for safepoints.
-        // -XX:LoopStripMiningIter=0 and -XX:LoopUnrollLimit=0 - are
-        // used to prevent optimizations over the loop in test_loop()
-        // since we actually want it to provoke a safepoint timeout.
-        // -XX:-UseBiasedLocking - is used to prevent biased locking
-        // handshakes from changing the timing of this test.
+    public static void testWith(int sfpt_interval, int timeout_delay, int unsafe_wait) throws Exception {
         ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                "-Xbootclasspath/a:.",
                 "-XX:+UnlockDiagnosticVMOptions",
-                "-XX:-UseBiasedLocking",
+                "-XX:+WhiteBoxAPI",
                 "-XX:+SafepointTimeout",
                 "-XX:+SafepointALot",
                 "-XX:+AbortVMOnSafepointTimeout",
                 "-XX:SafepointTimeoutDelay=" + timeout_delay,
                 "-XX:GuaranteedSafepointInterval=" + sfpt_interval,
-                "-XX:-TieredCompilation",
-                "-XX:-UseCountedLoopSafepoints",
-                "-XX:LoopStripMiningIter=0",
-                "-XX:LoopUnrollLimit=0",
-                "-XX:CompileCommand=compileonly,TestAbortVMOnSafepointTimeout::test_loop",
-                "-Xcomp",
                 "-XX:-CreateCoredumpOnCrash",
                 "-Xms64m",
                 "TestAbortVMOnSafepointTimeout",
-                "runTestLoop"
+                "" + unsafe_wait
         );
 
         OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        output.shouldMatch("Timed out while spinning to reach a safepoint.");
         if (Platform.isWindows()) {
             output.shouldMatch("Safepoint sync time longer than");
         } else {
             output.shouldMatch("SIGILL");
             if (Platform.isLinux()) {
                 output.shouldMatch("(sent by kill)");
             }
-            output.shouldMatch("TestAbortVMOnSafepointTimeout.test_loop");
         }
         output.shouldNotHaveExitValue(0);
```

Comment:
Looks like the test doesn't require that this mesg get printed:
`System.out.println("This message would occur after some time.");`

And it is set up to detect that the SafepointTimeout happened
which is what we want the test to verify at the core.

Reply:
The line "This message would occur after some time." should never be print if VM is working.
If the VM fails for some reason and the timeout is not performed, line:
"Timed out while spinning to reach a safepoint." is never printed and the OutputAnalyzer fails the test.
If we did timeout and it was printed we know that we didn't print the other message, since the only thread that can timeout is the one printing that message.

The second part verifies that the SIGILL was delivered.

Reply:
Okay, but then this message when you're reading the code is misleading:
`System.out.println("This message would occur after some time.");`
It should be printing something like:
`System.out.println("This message only prints if something is broken.");`

Reply:
I removed comment in last update, since it can't be printed.